examples/attention_is_all_you_need.py

```diff
@@ -163,79 +221,51 @@ def main(nH=6, dropout=0.1, nS=6, nB=15, nE=20, use_gpu=-1, lim=2000):
     train_X, train_Y = zip(*train)
     dev_X, dev_Y = zip(*dev)
     test_X, test_Y = zip(*test)
-    X_positions = Model.ops.position_encode(MAX_LENGTH, MODEL_SIZE)
-    print('Position encodings computed successfully')
     ''' Read dataset '''
     nlp_en = spacy.load('en_core_web_sm')
     nlp_de = spacy.load('de_core_news_sm')
     print('Models loaded')
-    # eos_vector = Model.ops.xp.random.rand(MODEL_SIZE)
-    # bos_vector = Model.ops.xp.random.rand(MODEL_SIZE)
-    # resize_vectors(nlp_de.vocab.vectors)
-    # nlp_de.vocab.set_vector('<eos>', eos_vector)
-    # nlp_de.vocab.set_vector('<bos>', bos_vector)
-    en_tokenizer = English().Defaults.create_tokenizer(nlp_en)
-    de_tokenizer = German().Defaults.create_tokenizer(nlp_de)
-    train_X, train_Y = spacy_tokenize(en_tokenizer, de_tokenizer,
-                                      train_X[:lim],
-                                      train_Y[:lim], MAX_LENGTH)
-    dev_X, dev_Y = spacy_tokenize(en_tokenizer,
-                                  de_tokenizer,
-                                  dev_X[:lim], dev_Y[:lim], MAX_LENGTH)
-    test_X, test_Y = spacy_tokenize(en_tokenizer, de_tokenizer,
-                                    test_X[:lim], test_Y[:lim], MAX_LENGTH)
-    train_Y = add_eos_bos(train_Y)
-    # link_vectors_to_models(nlp_en.vocab)
-    # link_vectors_to_models(nlp_de.vocab)
-    # en_embeddings = StaticVectors(nlp_en.vocab.vectors.name, MODEL_SIZE, column=0)
-    # de_embeddings = StaticVectors(nlp_de.vocab.vectors.name, MODEL_SIZE, column=0)
-    en_word2indx, en_indx2word = numericalize_vocab(nlp_en)
-    de_word2indx, de_indx2word = numericalize_vocab(nlp_de)
-    ''' 1 for oov, 1 for pad '''
-    nTGT = len(de_word2indx) + 2
-    en_embeddings = Embed(MODEL_SIZE, nM=MODEL_SIZE, nV=nTGT)
-    de_embeddings = Embed(MODEL_SIZE, nM=MODEL_SIZE, nV=nTGT)
-    model = EncoderDecoder(nH=nH, nS=nS, nTGT=nTGT)
-
+    for control_token in ("<eos>", "<bos>", "<pad>"):
+        nlp_en.tokenizer.add_special_case(control_token, [{ORTH: control_token}])
+        nlp_de.tokenizer.add_special_case(control_token, [{ORTH: control_token}])
+    train_X, train_Y = spacy_tokenize(nlp_en.tokenizer, nlp_de.tokenizer,
+                                      train_X[-lim:], train_Y[-lim:], MAX_LENGTH)
+    dev_X, dev_Y = spacy_tokenize(nlp_en.tokenizer, nlp_de.tokenizer,
+                                  dev_X[-lim:], dev_Y[-lim:], MAX_LENGTH)
+    test_X, test_Y = spacy_tokenize(nlp_en.tokenizer, nlp_de.tokenizer,
+                                    test_X[-lim:], test_Y[-lim:], MAX_LENGTH)
+    set_numeric_ids(nlp_en.vocab, train_X, vocab_size=VOCAB_SIZE,
```

Comment:
Is there any reason we are not forcing also the `<pad>` token? It is already set from somewhere?